### DISCUSSION TITLE
Filesystem locks in FSObject

### DISCUSSION THREAD

**Discussion by tristanexsquare** [minio/minio#12040] [Q&A] [answered]
We are currently rewriting some parts of the FSObject implementation, but we are struggling to understand why FSObject uses NSLock at ObjectLayer level operations and additionally the filesystem locks with rwPool. Can someone explain the motivation of the additional filesystem locking with rwPool. Could it be because of some writes to the metadata during ObjectLayer read operations?

**Comment by harshavardhana** [comment, marked as answer]
Because we use filesystem locks to run multiple instances on top of same backend.

This is a requirement to run multiple instances on a NAS mount. 

Changing all of this will lead to bugs and you won't know what you are doing. Don't do it.

**Reply by iternity-dotcom** [reply]
But they are only applied to the meta data, right? 
So the user data is not leveraging filesystem locks in such a scenario, correct?
